src/app/components/pages/MyProgress.tsx

```diff
@@ -37,8 +38,44 @@ const statistics = {
   tagColWidth: "col-lg-12",
 };
 
-interface MyProgressProps extends RouteComponentProps<{ userIdOfInterest: string }> {
+const QuestionParts = ({ progress }: { progress: MyProgressState | undefined }) => {
+  return (
+    <div className="mt-4">
+      <h4>Question parts correct by Type</h4>
+      <Row>
+        {statistics.questionTypeStatsList.map((qType: string) => {
+          const correct = progress?.correctByType?.[qType] ?? null;
+          const attempts = progress?.attemptsByType?.[qType] ?? null;
+          const percentage = safePercentage(correct, attempts);
+          return (
+            <Col key={qType} className={`${statistics.typeColWidth} mt-2 type-progress-bar`}>
+              <div className={"px-2"}>{HUMAN_QUESTION_TYPES[qType]} questions correct</div>
+              <div className={"px-2"}>
+                <ProgressBar percentage={percentage ?? 0}>
+                  {percentage == null ? "No data" : `${correct} of ${attempts}`}
+                </ProgressBar>
+              </div>
+            </Col>
+          );
+        })}
+      </Row>
+    </div>
+  );
+};
+
+const canCurrentUserViewOtherUserData = ({
```

Comment:
I think the logic inside this function can be simplified by making the return type a `boolean` (which I believe was your intention anyway).

Reply:
So this one was actually somewhat deliberate (though your comment made me catch a different problem :D). Basically I was trying to achieve 3 possible values for this; if true it's a teacher who is trying to view other users, if false it's a non-teacher trying to view other users (which in turn results in <Unauthorised> being returned). If undefined it's a non-teacher trying to view their own data so will behave accordingly.  There might be a better way of handling it - it was working before, but sonarLint didn't like the complexity, so I tried to extract the logic a bit, though I'm not convinced by it.

Reply:
That makes sense, but I would argue that given the function name, if I want to use this function I only care about whether the current user can view other user data or not. So there are only 2 possible values that you're interested about: `true` or `false`. If there is a third value, then I would expect this function to be named differently, or to have a different return type.

Reply:
I had a rethink on this one and did a quick refactor this morning

Reply:
It looks a lot better, thanks!